staging/src/k8s.io/apiserver/pkg/endpoints/filterlatency/filterlatency.go

```diff
@@ -57,7 +59,11 @@ func TrackStarted(handler http.Handler, name string) http.Handler {
 // it updates the corresponding metric with the filter latency duration.
 func TrackCompleted(handler http.Handler) http.Handler {
 	return trackCompleted(handler, utilclock.RealClock{}, func(ctx context.Context, fr *requestFilterRecord, completedAt time.Time) {
-		metrics.RecordFilterLatency(ctx, fr.name, completedAt.Sub(fr.startedTimestamp))
+		latency := completedAt.Sub(fr.startedTimestamp)
+		metrics.RecordFilterLatency(ctx, fr.name, latency)
+		if klog.V(3).Enabled() {
```

Comment:
We can do something like if klog.V(3).Enabled() && latency > slowFilterLatencyThreshold {
  httplog.AddInfof()
}

This will significantly reduce size of log for happy requests and improve readability. 

I suggest using slowFilterLatencyThreshold = 100ms?

Reply:
+1 for doing only if slow - it would be consistent with how we trace:
https://github.com/kubernetes/utils/blob/master/trace/trace.go

[I don't have strong opinion about the value itself.]